app/containers/Header/Header.jsx

```diff
@@ -11,55 +12,152 @@ class Header extends React.Component {
     super(props)
 
     this.state = {
+      collectionsInDrawer: true,
+      isDrawerOpen: false,
       isUserMenuOpen: false
     }
 
+    this.outerRef = React.createRef()
+    this.shadowRef = React.createRef()
+
+    this.closeDrawer = () => this.setState({isDrawerOpen: false})
     this.closeUserMenu = () => this.setState({isUserMenuOpen: false})
+
+    this.toggleDrawer = () =>
+      this.setState(({isDrawerOpen}) => ({isDrawerOpen: !isDrawerOpen}))
     this.toggleUserMenu = () =>
       this.setState(({isUserMenuOpen}) => ({isUserMenuOpen: !isUserMenuOpen}))
   }
 
+  componentDidMount() {
+    this.setDrawer()
+  }
+
+  componentDidUpdate() {
+    this.setDrawer()
+  }
+
+  setDrawer() {
+    if (!this.outerRef.current || !this.shadowRef.current) return
+
+    const shouldBeInDrawer =
+      this.outerRef.current.clientWidth <= this.shadowRef.current.clientWidth
+
+    if (!shouldBeInDrawer && this.state.isDrawerOpen) {
+      this.setState({isDrawerOpen: false})
+    }
+
+    if (shouldBeInDrawer !== this.state.collectionsInDrawer) {
+      this.setState({collectionsInDrawer: shouldBeInDrawer})
+    }
+  }
+
   render() {
     const {
       app: {
         config: {
+          api,
           whitelabel: {displayVersionNumber, logo}
         },
         version
       },
+      route: {
+        params: {collection: collectionName}
+      },
       user: {
-        isSignedIn,
         remote: {clientId, data}
       }
     } = this.props
-    const {isUserMenuOpen} = this.state
-
-    if (!isSignedIn) {
-      return null
-    }
-
+    const {collectionsInDrawer, isDrawerOpen, isUserMenuOpen} = this.state
+    const menuItems = buildGroupedMenuItems(api, collectionName)
     const displayName = (data && data.publishFirstName) || clientId
 
-    const userMenuStyle = new Style(styles, 'user-menu-wrapper').addIf(
+    const userMenuWrapperStyle = new Style(styles, 'user-menu-wrapper').addIf(
       'open',
       isUserMenuOpen
     )
 
     return (
       <header className={styles.header}>
+        <div className={styles['collections-wrapper']} ref={this.outerRef}>
+          {collectionsInDrawer ? (
+            <div className={styles['drawer-wrapper']}>
+              <button
+                className={styles['drawer-toggle']}
+                onClick={this.toggleDrawer}
+              >
+                <i className="material-icons" id={styles['menu-icon']}>
```

Comment:
This will still be replaced by the new component for icons, right?

Reply:
Yes